packages/flame/lib/src/components/blueprint.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/foundation.dart';
+
+import '../../components.dart';
+import '../game/flame_game.dart';
+
+const _attachedErrorMessage = "Can't add to attached Blueprints";
+
+/// A [Blueprint] is a virtual way of grouping [Component]s
+/// that are related, but they need to be added directly on
+/// the [FlameGame] level.
+abstract class Blueprint {
```

Comment:
Could this be a component with `HasGameRef`? But instead of having children, when you add components to it it adds them to the game instance?

Reply:
Yes, this was my first idea, but it bothered me to game a component added to the game tree, that would not be doing anything else once it have loaded.

Reply:
I think it is better to keep it as a `Component` than to add another `add` method to `FlameGame` when we have just managed to remove all of the other ones from the components (like `addHitbox` and `addEffect`).
It could be useful to put logic in its `update` method that is related to the group too, for example logic relating to the whole house (in the example) and not only specific parts of the house.

Reply:
Idk, it feels super weird to me to have a component that its purpose is to call `gameRef.add`, also, it could be confusing IMO, imagine we create `House` component instead that adds the components to the game, if you remove the `House` component, it would not remove "its" children.

By having a group that is not inside the game tree we make a statement that this is a virtual grouping, and we could even add more methods to it to help managing this group, like a remove method for example, that would call `game.remove` for all its components.

Reply:
I don't think it would be intuitive to do this by a component, the user would have to create a class based on `Component`, add HasGameRef for it, etc, etc. We could provide a component for it maybe, like a `BlueprintComponent` that implements this concept using a component and maybe this would be more straight forward, but this would still have a children attribute that wouldn't be used and etc, that is what kind makes me think that using a component to solve this is a overkill.

Reply:
also if it's a component you can use Blueprints on any level of the tree, for example you might want to have [Player, HouseTop, HouseBottom] as sibling but within the Stage component, not the root. or camera component, etc

Reply:
> We could provide a component for it maybe, like a `BlueprintComponent` that implements this concept using a component and maybe this would be more straight forward

Yes! That is my suggestion, I think it would improve this solution a lot. :)

Reply:
> Yes! That is my suggestion, I think it would improve this solution a lot. :)

Don't you think this could be hard to track? We would have to override everything related to children on this component so it would access the virtual components instead of its own components.

Reply:
> Don't you think this could be hard to track? We would have to override everything related to children on this component so it would access the virtual components instead of its own components.

I think that it is only `add`, `addAll`, `remove` and `removeAll` that we need to override.
They will probably stay the same for some time now (I hope), feels like the `Component` API has stabilized quite well.

Reply:
Only `add` and `remove`.*
(Since the others just proxy the calls to `add` and `remove`).